This command only works in a clone of flowtype/flow-typed. It is a tool used to validate the library definitions flow-typed project.

I have been trying to enhance some of the type definitions but running on windows does not seem to be fully supported and I was not able to run the required unit tests.

I started by creating a `run_def_tests.cmd` version of `run_def_tests.sh` that looks like this:
```
cd definitions
call npm install
call npm test
cd ..\cli
call npm install 
node dist\cli.js validate-defs 
node dist\cli.js run-tests %*
```

and it seems to work but:

1) when running `eslint npm/**/test*` I get a large number of warning and an error:
```
D:\MyDev\doberkofler\flow-typed\definitions\npm\react-redux_v4.x.x\flow_v0.30.x-\test_Provider.js
  6:11  error  Parsing error: Unexpected token, expected ,
```
Is this expected?

2) Later when processing the script, I get the following error message:
```
D:\MyDev\doberkofler\flow-typed\cli>node dist\cli.js validate-defs
This command only works in a clone of flowtype/flow-typed. It is a tool used to validate the library definitions flow-typed project.
```
clearly I am working in a cloned repository and I have no idea what this is about?

3) More generally speaking, I'm surprised why there are so many open pull requests and was wondering, if this is just a backlog or something else?

--- a/cli/src/commands/runTests.js
+++ b/cli/src/commands/runTests.js
@@ -87,6 +87,10 @@ async function getOrderedFlowBinVersions(): Promise<Array<string>> {
       const QUERY_PAGE_SIZE = 100;
       const OS_ARCH_FILTER_RE = new RegExp(BIN_PLATFORM);
 
+      if (!await fs.exists(BIN_DIR)) {
+        await fs.mkdir(BIN_DIR);
+      }
+
       let binURLs = new Map();
       let apiPayload = null;
       let page = 0;
@@ -132,10 +136,6 @@ async function getOrderedFlowBinVersions(): Promise<Array<string>> {
       FLOW_BIN_VERSION_ORDER.sort((a, b) => {
         return semver.lt(a, b) ? -1 : 1;
       });
-
-      if (!await fs.exists(BIN_DIR)) {
-        await fs.mkdir(BIN_DIR);
-      }
 
       await P.all(Array.from(binURLs).map(async ([version, binURL]) => {
         const zipPath = path.join(BIN_DIR, "flow-" + version + ".zip");

--- a/cli/src/lib/isInFlowTypedRepo.js
+++ b/cli/src/lib/isInFlowTypedRepo.js
@@ -1,5 +1,8 @@
 // @flow
 
+import * as node_path from 'path';
+
 export default function isInFlowTypedRepo() {
-  return /\/flow-typed/.test(process.cwd());
+  const PATTERN = new RegExp('\\' + node_path.sep + 'flow-typed');
+  return PATTERN.test(process.cwd());
 }

--- a/run_def_tests.cmd
+++ b/run_def_tests.cmd
@@ -1,0 +1,8 @@
+cd definitions
+call npm install
+call npm test
+cd ..\cli
+call npm install
+node dist\cli.js validate-defs ..\definitions
+node dist\cli.js run-tests %*
+cd ..

